ci/utils/pr_test_summary.py

```
@@ -0,0 +1,235 @@
+    return {
+        "Authorization": f"Bearer {token}",
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+
+
+def _paginate(path, token):
+    """Yield all items from a paginated GitHub REST API GET endpoint."""
+    url = f"https://api.github.com{path}?per_page=100"
+    while url:
+        req = urllib.request.Request(url, headers=_headers(token))
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)